suc/pkg/service/config.go

```diff
@@ -0,0 +1,137 @@
+package service
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/rancher/wins/cmd/server/config"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	ConfigDirEnvVar      = envVar{varName: "wins_config_dir"}
+	DebugEnvVar          = envVar{varName: "debug"}
+	AgentStringTLSEnvVar = envVar{varName: "strict_verify"}
+)
+
+const (
+	defaultConfigFile = "C:/etc/rancher/wins/config"
+)
+
+type envVar struct {
+	varName string
+}
+
+// get returns the value of the environment variable
+// using os.Getenv. The full name is passed to os.Getenv
+// (i.e. CATTLE_WINS_*)
+func (e *envVar) get() string {
+	return os.Getenv(e.name())
+}
+
+// getSimple returns the value of the environment variable
+// using os.Getenv. The simple name is passed to os.Getenv
+// (i.e. there is no 'CATTLE_WINS_' prefix)
+func (e *envVar) getSimple() string {
+	return os.Getenv(e.simpleName())
+}
+
+// name returns the name of the environment variable, including
+// a 'CATTLE_WINS_' prefix
+func (e *envVar) name() string {
+	return fmt.Sprintf("CATTLE_WINS_%s", strings.ToUpper(e.varName))
+}
+
+// simpleName returns the name of the environment variable, without
+// including the 'CATTLE_WINS_' prefix
+func (e *envVar) simpleName() string {
+	return strings.ToUpper(e.varName)
+}
+
+// loadConfig utilizes the ConfigDirEnvVar environment variable and path parameter to load
+// a config file located on the host. If both are provided, the path parameter will take
+// precedence. If neither are provided, then the defaultConfigFile path is used.
+func loadConfig(path string) (*config.Config, error) {
+	configDirEnv := os.Getenv(ConfigDirEnvVar.get())
+	if path == "" && configDirEnv != "" {
+		path = configDirEnv
+	}
+	if path == "" {
+		path = defaultConfigFile
+	}
```

Comment:
IMO this could probably be it's own function like `getConfigDir()`

Reply:
Fair, added `getConfigPath`